### PR TITLE
Create prod environment

### PR DESCRIPTION
**Pull request by calvinnhieu**
- terraform resources
- update functional tests to include prod
- update deploy scripts

### CODE REVIEW

File: .gitlab-ci-prod.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  - apt-get -y update
+  - apt-get -y install jq moreutils gettext unzip zip
+  - pip install -r requirements-dev.txt
+  - mkdir ./external_binaries
+  - curl https://releases.hashicorp.com/terraform/0.11.10/terraform_0.11.10_linux_amd64.zip -o /tmp/terraform.zip
+  - unzip /tmp/terraform.zip -d ./external_binaries/
+  - rm /tmp/terraform.zip
+  - export PATH=$PATH:`pwd`/external_binaries/
+  - export DEPLOYMENT_STAGE=$CI_COMMIT_REF_NAME
+  - if [ "$DEPLOYMENT_STAGE" == "master" ]; then
```
Comment (parthshahva) [inline .gitlab-ci-prod.yml]: Why do we need references to the other environments in gitlab file for prod?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ before_script:
   - export DEPLOYMENT_STAGE=$CI_COMMIT_REF_NAME
   - if [ "$DEPLOYMENT_STAGE" == "master" ]; then
   -   export DEPLOYMENT_STAGE=dev
-  - elif [[ "$DEPLOYMENT_STAGE" != "integration" && "$DEPLOYMENT_STAGE" != "staging" ]]; then
+  - elif [[ "$DEPLOYMENT_STAGE" != "integration" && "$DEPLOYMENT_STAGE" != "staging" && "DEPLOYMENT_STAGE" != "prod" ]]; then
```
Comment (parthshahva) [inline .gitlab-ci.yml]: Why do we need reference to prod in non prod gitlab file?

File: .gitlab-ci-prod.yml (hunk 1 above)
Comment (calvinnhieu) [reply .gitlab-ci-prod.yml]: Removing

File: chalice/.chalice/config.json [hunk 3]
```diff
@@ -56,6 +56,23 @@
         "DYNAMO_OUTPUT_TABLE_NAME": "dcp-matrix-service-output-table-predev",
         "S3_RESULTS_BUCKET": "dcp-matrix-service-results-predev"
       }
+    },
+    "prod": {
+      "api_gateway_stage": "prod",
+      "environment_variables": {
+        "DEPLOYMENT_STAGE": "prod",
+        "DSS_STAGE": "prod",
+        "MATRIX_VERSION": "52940c5",
```
Comment (parthshahva) [inline chalice/.chalice/config.json]: It'd be nice if we only deployed tags to prod

File: terraform/modules/matrix-service/infra/converter_batch.tf [hunk 4]
```diff
@@ -108,7 +108,7 @@ resource "aws_batch_compute_environment" "converter_compute_env" {
     // You must set desired_vcpus otherwise you get error: "desiredvCpus should be between minvCpus and maxvCpus"
     // However this is actually not settable in AWS.  It will not let you change it.
     // Here we use an external data source to dynamically set the desired vcpus to match current state.
-    desired_vcpus = "${data.external.converter_desired_vcpus.result.desired_vcpus}"
+    desired_vcpus = 4
```
Comment (parthshahva) [inline terraform/modules/matrix-service/infra/converter_batch.tf]: Why this change? This will cause terraform to fail when the set desired vcpus does not match 4? We should keep the external data source reference

Comment (calvinnhieu) [reply terraform/modules/matrix-service/infra/converter_batch.tf]: Woops meant to change that back. That was a workaround since the script wasn't working for me.

File: chalice/.chalice/config.json (hunk 3 above)
Comment (calvinnhieu) [reply chalice/.chalice/config.json]: For prod deployments, tagging with SEMVER version e.g. `v0.0.1`

File: .gitlab-ci.yml (hunk 2 above)
Comment (calvinnhieu) [reply .gitlab-ci.yml]: Removing

File: terraform/modules/matrix-service/infra/converter_batch.tf (hunk 4 above)
Comment (calvinnhieu) [reply terraform/modules/matrix-service/infra/converter_batch.tf]: Will investigate script after prod is green

File: matrix/common/dss_zarr_store.py [hunk 5]
```diff
@@ -90,6 +92,26 @@ def __init__(self, bundle_uuid, bundle_version=None, replica="aws"):
...
+
+        if deployment_stage == "prod":
+            dss_config['DSSClient']['swagger_url'] = "https://dss.data.humancellatlas.org/v1/swagger.json"
+        elif deployment_stage == "staging":
+            dss_config['DSSClient']['swagger_url'] = f"https://dss.staging.data.humancellatlas.org/v1/swagger.json"
+        else:
+            dss_config['DSSClient']['swagger_url'] = f"https://dss.integration.data.humancellatlas.org/v1/swagger.json"
+
+        client = hca.dss.DSSClient(config=dss_config)
+        logger.debug(f"Initailized DSSClient with host URL {client.host}")
```
Comment (mckinsel) [inline matrix/common/dss_zarr_store.py]: Initialized

File: matrix/common/dss_zarr_store.py [hunk 6]
```diff
@@ -49,14 +50,15 @@ class DSSZarrStore(MutableMapping):
     def _transform_key(self, key):
         return key.replace("/", self._separator_char)
 
-    def __init__(self, bundle_uuid, bundle_version=None, replica="aws"):
+    def __init__(self, request_hash, bundle_uuid, bundle_version=None, replica="aws"):
```
Comment (mckinsel) [inline matrix/common/dss_zarr_store.py]: Requiring `request_hash` seems to add in a new, unrelated concern to this class.

Comment (calvinnhieu) [reply matrix/common/dss_zarr_store.py]: I see your point and am OK with removing it. The goal was just to improve the logs by including the hash

Comment (mckinsel) [reply matrix/common/dss_zarr_store.py]: Gotcha. At some point we're going to have to give some code to people who want to read matrices from bundles directly. Ideally this class should just do it, along with the rest of the zarr package. So I'm reluctant to add in tighter coupling with matrix service-specific stuff.

Comment (calvinnhieu) [reply matrix/common/dss_zarr_store.py]: Removed
